pyuoi/linear_model/base.py

```
@@ -532,15 +532,17 @@ def __init__(self, n_boots_sel=24, n_boots_est=24, selection_frac=0.9,
 
     def _pre_fit(self, X, y):
         X, y = super()._pre_fit(X, y)
+        if y.ndim == 1:
+            y = y[:, np.newaxis]
+        elif y.ndim == 2:
+            if y.shape[1] > 1:
+                raise ValueError('y should either have shape ' +
+                                 '(n_samples, ) or (n_samples, 1).')
```

This should also raise an error if `y.ndim > 2`.